cmd/datasource-syncer/main.go

```diff
@@ -127,35 +117,30 @@ func main() {
 		dataSource, err := grafanaClient.DataSourceByUID(datasourceUID)
 		if err != nil {
 			dsErrors = append(dsErrors, datasourceUID)
-			//nolint:errcheck
-			level.Error(logger).Log("msg", fmt.Sprintf("error fetching data source config of data source uid: %s", datasourceUID), "err", err)
+			logger.Error(fmt.Sprintf("error fetching data source config of data source uid: %s", datasourceUID), "err", err)
 			continue
 		}
 
 		dataSource, err = buildUpdateDataSourceRequest(*dataSource, token)
 		if err != nil {
 			dsErrors = append(dsErrors, datasourceUID)
-			//nolint:errcheck
-			level.Error(logger).Log("msg", fmt.Sprintf("couldn't build data source update request for data source uid: %s", datasourceUID), "err", err)
+			logger.Error(fmt.Sprintf("couldn't build data source update request for data source uid: %s", datasourceUID), "err", err)
 			continue
 		}
 
 		err = grafanaClient.UpdateDataSourceByUID(dataSource)
 		if err != nil {
 			dsErrors = append(dsErrors, datasourceUID)
-			//nolint:errcheck
-			level.Error(logger).Log("msg", fmt.Sprintf("couldn't send update data source request to data source id: %s", datasourceUID), "err", err)
+			logger.Error(fmt.Sprintf("couldn't send update data source request to data source id: %s", datasourceUID), "err", err)
 			continue
 		}
 		dsSuccessfullyUpdated = append(dsSuccessfullyUpdated, datasourceUID)
 	}
 	if len(dsSuccessfullyUpdated) != 0 {
-		//nolint:errcheck
-		level.Info(logger).Log("msg", fmt.Sprintf("Updated Grafana data source uids: %s", dsSuccessfullyUpdated))
+		logger.Info(fmt.Sprintf("Updated Grafana data source uids: %s", dsSuccessfullyUpdated))
 	}
 	if len(dsErrors) != 0 {
-		//nolint:errcheck
-		level.Error(logger).Log("msg", fmt.Sprintf("Failed to update Grafana data source uids: %s", dsErrors))
+		logger.Error(fmt.Sprintf("Failed to update Grafana data source uids: %s", dsErrors))
```

Comment:
This can fixed to eliminate the `fmt.Sprintf()`.

```suggestion
		logger.Error("Failed to update Grafana data source uids", "error", dsErrors)
```